File: Real World Imperative Programming CSharp/FileEntry.cs
namespace RealWorldImperativeProgrammingCSharp
{
    public class FileEntry
    {
        private readonly string path;

        public FileEntry(string path)
        {
            this.path = path;
        }

        public string Path
        {
            get
            {
                return path;
            }
        }

        public string Name
        {
            get
            {
                return System.IO.Path.GetFileName(path);
            }
        }
    }
}

File: Real World Functional Programming FSharp/FileEntry.fs
namespace RealWorldFunctionalProgrammingFSharp

open System
open System.IO


type FileEntry = { Path: string; Name: string }

[<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
module FileEntry =
   
    let create path = { Path = path; Name = Path.GetFileName path } 
   
    let empty = create String.Empty       
